src/components/table/_macro.njk

```diff
@@ -17,55 +18,111 @@
                 </caption>
             {% endif %}
             <thead class="ons-table__head">
-                <tr class="ons-table__row">
-                    {% for th in params.ths %}
-                        <th
-                            scope="col"
-                            class="ons-table__header{{ ' ' + th.thClasses if th.thClasses else '' }}{{ ' ons-table__header--numeric'if th.numeric }}"
-                            {% if 'sortable' in variants %}aria-sort="{{- th.ariaSort | default('none') -}}"{% endif %}{% if th.widthPercentage %}width="{{ th.widthPercentage }}%"{% endif %}
-                        >
-                            <span class="ons-table__header-text">{{- th.value -}}</span>
-                            {% if 'sortable' in variants %}
-                                {{-
-                                    onsIcon({
-                                        "iconType": "sort-sprite",
-                                        "id": th.value | replace(' ', '-'),
-                                        "classes": 'ons-u-d-no'
-                                    })
-                                -}}
-                            {% endif %}
-                        </th>
+                {% if params.thList %}
+                    {% for header_cols in params.thList %}
+                        <tr class="ons-table__row">
+                            {% for th in header_cols.ths %}
+                                <th
+                                    scope="col"
+                                    class="ons-table__header{{ ' ' + th.thClasses if th.thClasses else '' }}{{ ' ons-table__header--numeric'if th.numeric }}"
+                                    {% if th.colspan %}colspan="{{ th.colspan }}"{% endif %}
+                                    {% if th.rowspan %}rowspan="{{ th.rowspan }}"{% endif %}
+                                    {% if 'sortable' in variants %}aria-sort="{{- th.ariaSort | default('none') -}}"{% endif %}
+                                    {% if th.widthPercentage %}width="{{ th.widthPercentage }}%"{% endif %}
+                                >
+                                    <span class="ons-table__header-text">{{- th.value | safe -}}</span>
+                                    {% if 'sortable' in variants %}
+                                        {{-
+                                            onsIcon({
+                                                "iconType": "sort-sprite",
+                                                "id": th.value | replace(' ', '-'),
+                                                "classes": 'ons-u-d-no'
+                                            })
+                                        -}}
+                                    {% endif %}
```

Comment:
This seems to be duplicated, you could restructure this logic so that it only needs to be used once